### PR TITLE
Update PHP version, phpunit and testcases for Plates v3 (#257)

### LINKED ISSUE #257: Php 8 compatibility
Could you update the code of the stable version of the template engine for compatibility with php 8 and composer 2?

### PR DESCRIPTION
**Pull request by roxblnfk**
I've updated devtools and test cases.
Phpunit now works with PHP ^7.4|^8.0.
However, the template engine code is still compatible with PHP 5.3.
I ran tests on PHP 7.4 and PHP 8.0. All tests passed.

Fix #257

### CODE REVIEW

**Comment by ragboyjr** [comment]
Exciting! Will take a look this eve!

File: src/Template/Name.php [hunk 1]
```diff
@@ -161,13 +161,17 @@ public function getFile()
     public function getPath()
     {
         if (is_null($this->folder)) {
-            return $this->getDefaultDirectory() . DIRECTORY_SEPARATOR . $this->file;
+            return "{$this->getDefaultDirectory()}/{$this->file}";
         }
 
-        $path = $this->folder->getPath() . DIRECTORY_SEPARATOR . $this->file;
+        $path = "{$this->folder->getPath()}/{$this->file}";
```
Comment (azjezz) [inline src/Template/Name.php:167]: should be reverted?

File: src/Template/Name.php [hunk 2]
```diff
@@ -161,13 +161,17 @@ public function getFile()
...
 
-        if (!is_file($path) and $this->folder->getFallback() and is_file($this->getDefaultDirectory() . DIRECTORY_SEPARATOR . $this->file)) {
-            $path = $this->getDefaultDirectory() . DIRECTORY_SEPARATOR . $this->file;
+        if (
+            !is_file($path)
+            && $this->folder->getFallback()
+            && is_file("{$this->getDefaultDirectory()}/{$this->file}")
+        ) {
+            $path = "{$this->getDefaultDirectory()}/{$this->file}";
```
Comment (azjezz) [inline src/Template/Name.php:174]: same here.

File: tests/EngineTest.php [hunk 3]
```diff
@@ -1,14 +1,15 @@
 <?php
 
-namespace League\Plates;
+namespace League\Plates\Tests;
```
Comment (azjezz) [inline tests/EngineTest.php:3]: is adding `Tests` necessary?

File: tests/EngineTest.php [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
 
+use League\Plates\Engine;
 use org\bovigo\vfs\vfsStream;
 
-class EngineTest extends \PHPUnit_Framework_TestCase
+class EngineTest extends \PHPUnit\Framework\TestCase
```
Comment (azjezz) [inline tests/EngineTest.php:8]: should probably be imported.

File: tests/EngineTest.php [hunk 5]
```diff
@@ -23,41 +24,42 @@ public function testCanCreateInstance()
...
         $this->assertInstanceOf('League\Plates\Engine', $this->engine->setDirectory(null));
-        $this->assertEquals($this->engine->getDirectory(), null);
+        $this->assertNull($this->engine->getDirectory());
     }
 
     public function testSetInvalidDirectory()
     {
-        $this->setExpectedException('LogicException', 'The specified path "vfs://does/not/exist" does not exist.');
+        // The specified path "vfs://does/not/exist" does not exist.
+        $this->expectException(\LogicException::class);
```
Comment (azjezz) [inline tests/EngineTest.php:39]: you can use `$this->expectExceptionMessage('...')` here, to ensure that the message is the same as the one expected.

File: tests/EngineTest.php [hunk 6]
```diff
@@ -71,19 +73,21 @@ public function testAddFolder()
...
         $this->assertInstanceOf('League\Plates\Engine', $this->engine->addFolder('folder', vfsStream::url('templates/folder')));
-        $this->assertEquals($this->engine->getFolders()->get('folder')->getPath(), 'vfs://templates/folder');
+        $this->assertSame('vfs://templates/folder', $this->engine->getFolders()->get('folder')->getPath());
     }
 
     public function testAddFolderWithNamespaceConflict()
     {
-        $this->setExpectedException('LogicException', 'The template folder "name" is already being used.');
+        // The template folder "name" is already being used.
+        $this->expectException(\LogicException::class);
```
Comment (azjezz) [inline tests/EngineTest.php:82]: same here.

File: tests/EngineTest.php [hunk 7]
```diff
@@ -71,19 +73,21 @@ public function testAddFolder()
...
         $this->engine->addFolder('name', vfsStream::url('templates'));
         $this->engine->addFolder('name', vfsStream::url('templates'));
     }
 
     public function testAddFolderWithInvalidDirectory()
     {
-        $this->setExpectedException('LogicException', 'The specified directory path "vfs://does/not/exist" does not exist.');
+        // The specified directory path "vfs://does/not/exist" does not exist.
+        $this->expectException(\LogicException::class);
```
Comment (azjezz) [inline tests/EngineTest.php:90]: same here.

File: tests/EngineTest.php [hunk 8]
```diff
@@ -139,20 +143,21 @@ public function testRegisterFunction()
...
         $this->engine->dropFunction('uppercase');
-        $this->assertEquals($this->engine->doesFunctionExist('uppercase'), false);
+        $this->assertFalse($this->engine->doesFunctionExist('uppercase'));
     }
 
     public function testDropInvalidFunction()
     {
-        $this->setExpectedException('LogicException', 'The template function "some_function_that_does_not_exist" was not found.');
+        // The template function "some_function_that_does_not_exist" was not found.
+        $this->expectException(\LogicException::class);
```
Comment (azjezz) [inline tests/EngineTest.php:160]: same here.

File: tests/Extension/AssetTest.php [hunk 9]
```diff
@@ -1,13 +1,17 @@
 <?php
 
-namespace League\Plates\Extension;
+declare(strict_types=1);
```
Comment (azjezz) [inline tests/Extension/AssetTest.php:3]: I think this should be removed. as it should be either applied everywhere in the project or simply nowhere.

and it cannot be applied everywhere as the PHP requirement from the project still includes 5.x

File: tests/Extension/AssetTest.php [hunk 10]
```diff
@@ -68,7 +72,8 @@ public function testCachedAssetUrlUsingFilenameMethod()
 
     public function testFileNotFoundException()
     {
-        $this->setExpectedException('LogicException', 'Unable to locate the asset "styles.css" in the "' . vfsStream::url('assets') . '" directory.');
+        // Unable to locate the asset "styles.css" in the @assets directory.
+        $this->expectException(\LogicException::class);
```
Comment (azjezz) [inline tests/Extension/AssetTest.php:76]: same as above.

File: tests/Extension/URITest.php [hunk 11]
```diff
@@ -1,14 +1,18 @@
 <?php
 
-namespace League\Plates\Extension;
+declare(strict_types=1);
```
Comment (azjezz) [inline tests/Extension/URITest.php:3]: same as above.

**Comment by ragboyjr** [comment]
This looks awesome!

**Pull request merged**